state/core/src/test/java/com/expediagroup/streamplatform/streamregistry/state/EntityViewUpdaterTest.java

```diff
@@ -18,68 +18,191 @@
+
+    val result = underTest.update(statusDeletionEvent);
+    assertThat(result, is(nullValue()));
+    assertThat(entities, hasEntry(key, deleted(null)));
+  }
+
+  @Test
+  public void deleteSpecificationEvent() {
+    val previousEntity = underTest.update(specificationEvent);
+
+    assertThat(previousEntity, is(nullValue()));
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, existing(entity.withStatus(oldStatus))));
+
+    val deletedEntity = underTest.update(specificationDeletionEvent);
+    assertThat(deletedEntity, is(entity.withStatus(oldStatus)));
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, deleted(entity.withStatus(oldStatus))));
+  }
+
+  @Test
+  public void deleteStatusEvent() {
+    entities.put(key, existing(oldEntity));
+
+    val updatedStatusEventResult = underTest.update(statusEvent);
+    assertThat(updatedStatusEventResult, is(oldEntity));
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, existing(oldEntity.withStatus(status))));
+
+    val deletedStatusEventResult = underTest.update(statusDeletionEvent);
+    assertThat(deletedStatusEventResult, is(oldEntity.withStatus(status)));
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, existing(oldEntity)));
+  }
+
+  @Test
+  public void deleteSpecificationEventWithNoPreviousEntity() {
+    val deletedEntity = underTest.update(specificationDeletionEvent);
+    assertThat(deletedEntity, is(nullValue()));
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, deleted(null)));
+  }
+
+  @Test
+  public void purgeNonExistingEntity() {
+    val purgedEntity = underTest.purge(key);
+
+    assertThat(purgedEntity.isPresent(), is(false));
+    assertThat(entities, is(aMapWithSize(0)));
+  }
+
+  @Test
+  public void purgeDeletedEntity() {
+    entities.put(key, deleted(oldEntity));
+
+    Optional<Entity<DomainKey, DefaultSpecification>> purgedEntity = underTest.purge(key);
+
+    assertThat(purgedEntity.isPresent(), is(true));
+    assertThat(purgedEntity.get(), is(oldEntity));
+    assertThat(entities, is(aMapWithSize(0)));
+  }
+
+  @Test
+  public void purgeNonDeletedEntity() {
+    entities.put(key, existing(oldEntity));
+
+    Optional<Entity<DomainKey, DefaultSpecification>> purgedEntity = underTest.purge(key);
+
+    assertThat(purgedEntity.isPresent(), is(false));
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, existing(oldEntity)));
+  }
+
+  @Test
+  public void purgeWithNullPreviousEntry() {
+    underTest.update(specificationDeletionEvent);
+    assertThat(entities, is(aMapWithSize(1)));
+    assertThat(entities, hasEntry(key, deleted(null)));
+
+    underTest.purge(key);
+
+    assertThat(entities, is(aMapWithSize(0)));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void unhandledEvents() {
```

Comment:
This is minor but the method name is quite vague. 
```suggestion
  public void updateWithNull() {
```